### PR TITLE
Slack terminal extension management

### PR DESCRIPTION
**Pull request by annetill**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the new behavior (if this is a feature change)?**
We are able to use and set SlackTerminal extension.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by annetill** [comment]
@geofjamg and @floriand-e2r I have just strated a WIP PR to illustrate the use of ````SlackTerminal```` extension. This PR is a draft and not completed at all. But it could be useful to validate the extension design.

**Comment by annetill** [comment]
@geofjamg this PR is not complete because are missing:
- To clean all slack terminal extensions of the synchronous component before the update ;
- To manage several slack terminal extensions present in the same synchronous component in the slack bus selector.
Sorry, I miss time (and mind) to finish this PR.

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowProvider.java [hunk 1]
```diff
@@ -106,11 +108,16 @@ private OpenLoadFlowParameters getParametersExt(LoadFlowParameters parameters) {
         return parametersExt;
     }
 
+    private static SlackBusSelector getSlackBusSelector(Network network, LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt) {
+        return parameters.isReadSlackBus() ? new NetworkSlackBusSelector(network, parametersExt.getSlackBusSelector())
+                                           : parametersExt.getSlackBusSelector();
```
Comment (mathbagu) [inline src/main/java/com/powsybl/openloadflow/OpenLoadFlowProvider.java:113]: I don't think the `isReadSlackBus` should be priority... I would say:
- if parametersExt.getSlackBusSelector is set, then use it
- if parameters.isReadSlackBus is set, then use NetworkSlackBusSelector
- default: use the default strategy

File: src/main/java/com/powsybl/openloadflow/network/NetworkSlackBusSelector.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
```
Comment (mathbagu) [inline src/main/java/com/powsybl/openloadflow/network/NetworkSlackBusSelector.java:2]: Fix copyright date

File: src/main/java/com/powsybl/openloadflow/network/NetworkSlackBusSelector.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public LfBus select(List<LfBus> buses) {
+        for (LfBus bus : buses) {
+            if (!bus.isFictitious() && slackBusIds.contains(bus.getId())) {
+                return bus;
+            }
+        }
+        // fallback to automatic selection
+        return fallbackSelector.select(buses);
+    }
+}
```
Comment (mathbagu) [inline src/main/java/com/powsybl/openloadflow/network/NetworkSlackBusSelector.java:58]: I don't know if it's work, but don't you have to check somewhere if the slack bus is in the right synchronous component

File: src/main/java/com/powsybl/openloadflow/network/NetworkSlackBusSelector.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/network/NetworkSlackBusSelector.java]: Done

File: src/main/java/com/powsybl/openloadflow/network/NetworkSlackBusSelector.java (hunk 3 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/network/NetworkSlackBusSelector.java:62]: Algo improved

**Pull request merged**